### PR TITLE
OC-544: Sets fields on response message.

### PR DESCRIPTION
**Pull request by robindenadel**

### CODE REVIEW

File: osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/application/services/DeviceResponseMessageService.java [hunk 1]
```diff
@@ -102,9 +102,9 @@ private boolean shouldRetryBasedOnException(final OsgpException e) {
             return false;
         }
 
-        final String exceptionMsg = e.toString();
+        String exceptionMsg = e.toString();
         if (e.getCause() != null) {
-            exceptionMsg.concat(e.getCause().toString());
+            exceptionMsg = exceptionMsg.concat(e.getCause().toString());
```
Comment (kevinsmeets) [inline osgp/platform/osgp-core/src/main/java/org/opensmartgridplatform/core/application/services/DeviceResponseMessageService.java]: What a silly mistake here... Good thing it is fixed 👍

File: osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/infra/messaging/DeviceRequestMessageListener.java [hunk 2]
```diff
@@ -93,7 +94,8 @@ private void sendNotSupportedException(final ObjectMessage objectMessage, final
                     .result(ResponseMessageResultType.NOT_OK)
                     .osgpException(osgpException)
                     .dataObject(objectMessage.getObject())
-                    .scheduled(false)
+                    .retryHeader(new RetryHeader())
+                    .scheduled(messageMetadata.isScheduled())
```
[the remark this answers is not in the document]
Comment (robindenadel) [reply osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/infra/messaging/DeviceRequestMessageListener.java]: OK

File: osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/messaging/BaseMessageProcessor.java [hunk 3]
```diff
@@ -100,7 +101,10 @@ public void checkForRedelivery(final DeviceMessageMetadata deviceMessageMetadata
                     deviceMessageMetadata.getOrganisationIdentification(),
                     deviceMessageMetadata.getDeviceIdentification(), deviceMessageMetadata.getCorrelationUid(),
                     deviceMessageMetadata.getMessagePriority());
-            this.handleExpectedError(deviceResponse, e, domainInformation, deviceMessageMetadata.getMessageType());
+            final Long scheduleTime = deviceMessageMetadata.getScheduleTime();
+            final boolean isScheduled = scheduleTime != null && scheduleTime > 0;
```
[the remark this answers is not in the document]
Comment (robindenadel) [reply osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/messaging/BaseMessageProcessor.java]: Done

File: osgp/protocol-adapter-oslp/signing-server/src/main/java/org/opensmartgridplatform/signing/server/application/services/SigningService.java [hunk 4]
```diff
@@ -77,23 +77,34 @@ private void doSignMessage(final UnsignedOslpEnvelopeDto unsignedOslpEnvelopeDto
...
-                    .withDeviceIdentification(deviceIdentification).withResult(ResponseMessageResultType.NOT_OK)
+                    .withDeviceIdentification(deviceIdentification)
+                    .withResult(ResponseMessageResultType.NOT_OK)
                     .withOsgpException(
                             new OsgpException(ComponentType.UNKNOWN, "Failed to build signed OslpEnvelope", null))
-                    .withDataObject(unsignedOslpEnvelopeDto).withMessagePriority(messagePriority).build();
+                    .withDataObject(unsignedOslpEnvelopeDto)
+                    .withMessagePriority(messagePriority)
+                    .withScheduled(scheduled)
+                    .withRetryHeader(new RetryHeader())
```
[the remark this answers is not in the document]
Comment (robindenadel) [reply osgp/protocol-adapter-oslp/signing-server/src/main/java/org/opensmartgridplatform/signing/server/application/services/SigningService.java]: removed the RetryHeader

File: osgp/protocol-adapter-oslp/signing-server/src/main/java/org/opensmartgridplatform/signing/server/application/services/SigningService.java [hunk 5]
```diff
@@ -102,10 +113,16 @@ private void doSignMessage(final UnsignedOslpEnvelopeDto unsignedOslpEnvelopeDto
...
+                    .withCorrelationUid(correlationUid)
                     .withOrganisationIdentification(organisationIdentification)
-                    .withDeviceIdentification(deviceIdentification).withResult(ResponseMessageResultType.OK)
-                    .withDataObject(signedOslpEnvelopeDto).withMessagePriority(messagePriority).build();
+                    .withDeviceIdentification(deviceIdentification)
+                    .withResult(ResponseMessageResultType.OK)
+                    .withDataObject(signedOslpEnvelopeDto)
+                    .withMessagePriority(messagePriority)
+                    .withScheduled(scheduled)
+                    .withRetryHeader(new RetryHeader())
```
[the remark this answers is not in the document]
Comment (robindenadel) [reply osgp/protocol-adapter-oslp/signing-server/src/main/java/org/opensmartgridplatform/signing/server/application/services/SigningService.java]: removed the RetryHeader

File: osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/messaging/BaseMessageProcessor.java [hunk 6]
```diff
@@ -129,14 +133,20 @@ public void handleDeviceResponse(final DeviceResponse deviceResponse,
...
-                .domain(domainInformation.getDomain()).domainVersion(domainInformation.getDomainVersion())
-                .deviceMessageMetadata(deviceMessageMetadata).result(result).osgpException(ex).retryCount(retryCount)
+                .domain(domainInformation.getDomain())
+                .domainVersion(domainInformation.getDomainVersion())
+                .deviceMessageMetadata(deviceMessageMetadata)
+                .result(result)
+                .osgpException(ex)
+                .retryCount(retryCount)
+                .retryHeader(new RetryHeader())
+                .scheduled(isScheduled)
```
[the remark this answers is not in the document]
Comment (robindenadel) [reply osgp/protocol-adapter-iec61850/osgp-protocol-adapter-iec61850/src/main/java/org/opensmartgridplatform/adapter/protocol/iec61850/infra/messaging/BaseMessageProcessor.java]: removed the RetryHeader

**Pull request merged**